gnosis/cowswap
# [presign] Post presign order

# Summary
The goal of this PR is to differentiate between orders that needs to pre-sign and not, in order to call the API.

* These ones that don't support offchain signing, will call the API as we used to: Adds the signature, and the relevant signature scheme
* The ones that don't will use presign. These order will:
   * call the API first, get the order id
   * Then, it will submit a transaction to presign the order (not included in this PR, it will be the next)

Additionally this PR does two things:
* Remove some legacy code for the signing schemes. Now, instead of redefining it, we can just take the enum from the NPM package
* For making it easy to test, this PR sets to false `allowsOffchainSigning` so EOAs will use presign. We can keep it like this until before we want to consolidate with develop.

# To Test

1. Connect EOA
2. Try to swap
3. The order will be sent and should remain pending ---> FOREVER (for now, no match will occur)
